Description: Fix a error about thread_local variables export to DLL, when compiled with clang
A related issue: https://github.com/drogonframework/drogon/issues/1845
LLM solution: use anonymous namespace to declare thread_local variables

--- a/lib/src/Utilities.cc	2026-10-12 15:44:34.000000000 +0800
+++ b/lib/src/Utilities.cc	2026-10-14 09:55:47.811615600 +0800
@@ -1329,6 +1329,12 @@
     return trantor::utils::secureRandomBytes(ptr, size);
 }
 
+namespace
+{
+thread_local trantor::utils::Hash256 hash;
+thread_local size_t i = 0;
+}  // namespace
+
 std::string secureRandomString(size_t size)
 {
     if (size == 0)
@@ -1345,8 +1351,6 @@
     // batch up to 32 bytes of random data for efficiency. Calling
     // secureRandomBytes can be expensive.
     auto randByte = []() {
-        thread_local trantor::utils::Hash256 hash;
-        thread_local size_t i = 0;
         if (i == 0)
         {
             bool ok = trantor::utils::secureRandomBytes(&hash, sizeof(hash));
